frontend/document/document.lua

```diff
@@ -39,6 +39,8 @@ function Document:new(from_o)
     self.__index = self
     if o._init then o:_init() end
     if o.init then o:init() end
+    -- Make sure the checksum has been generated before any document saving operations.
+    o:fastDigest()
```

Comment:
And as @chrox mentioned, this function is pretty cheap, it reads only 1024 * 10 bytes from a single file. You won't be able to feel any difference by enabling this line or not. And this change makes fastDigest function even cheap, it won't be calculated again.